ietf/bibliography/tests.py

```diff
@@ -0,0 +1,105 @@
+        self.home = home
+        self.client = client
+
+        self.rfc_2026 = RFC.objects.create(
+            name="draft-ietf-poised95-std-proc-3",
+            title="The Internet Standards Process -- Revision 3",
+            rfc="2026",
+        )
+
+        self.standard_index: StandardIndexPage = StandardIndexPageFactory(
+            parent=self.home,
+        )  # type: ignore
+
+        self.standard_page: StandardPage = StandardPageFactory(
+            parent=self.standard_index,
+        )  # type: ignore
+        self.standard_page.in_depth = [
+            {
+                "type": "raw_html",
+                "value": (
+                    f'<a data-app="snippets" data-id="{self.rfc_2026.pk}"'
+                    ' data-linktype="rfc">The Standards RFC</a>'
+                ),
+            }
+        ]
+        self.standard_page.save()
+
+    def test_bibliography_item_created(self):
+        assert BibliographyItem.objects.count() == 1
+        item = BibliographyItem.objects.get()
+        assert item.content_object == self.rfc_2026
+
+    def test_referenced_types(self, admin_client):
+        rfc_content_type = ContentType.objects.get_for_model(RFC)
+        response = admin_client.get(reverse("referenced_types"))
+        assert response.status_code == 200
+        html = response.content.decode()
+        assert reverse("referenced_objects", args=[rfc_content_type.pk]) in html
+        assert "snippets | RFC" in html
+
+    def test_referenced_objects(self, admin_client):
+        rfc_content_type = ContentType.objects.get_for_model(RFC)
+        response = admin_client.get(
+            reverse("referenced_objects", args=[rfc_content_type.pk])
+        )
+        assert response.status_code == 200
+        html = response.content.decode()
+        assert reverse(
+            "referencing_pages", args=[rfc_content_type.pk, self.rfc_2026.pk]
+        ) in html
+        assert "RFC 2026" in html
+
+    def test_referencing_pages(self, admin_client):
+        rfc_content_type = ContentType.objects.get_for_model(RFC)
+        response = admin_client.get(
+            reverse("referencing_pages", args=[rfc_content_type.pk, self.rfc_2026.pk])
+        )
+        assert response.status_code == 200
+        html = response.content.decode()
+        assert self.standard_page.title in html
+
+    def test_render_page(self, client):
+        response = client.get(self.standard_page.url)
+        assert response.status_code == 200
+        html = response.content.decode()
+        assert "RFC 2026" in html
+
+    def test_render_page_reference_removed(self, client):
+        self.rfc_2026.delete()
+        self.standard_page.save()
+        response = client.get(self.standard_page.url)
+        assert response.status_code == 200
+        html = response.content.decode()
+        assert "RFC 2026" not in html
+        assert "(removed)" in html
+
+    def test_update_fields_partial_raises_exception(self):
```

Comment:
I've added docstrings to this module, because it's not immediately obvious what it's supposed to do. Do you feel there are other tests that are worth documenting?

Reply:
Thanks for adding those docstrings :)

[Resolved]